# WELCOME
I'm Learning DataScience

### Importing Modules

In [2]:
import pandas as pd
import seaborn as sns

### Loading Titanic Data Set

In [3]:
df = sns.load_dataset("titanic")
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

### First Rows Of Data

In [4]:
df.head(7)

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   
5         0       3    male   NaN      0      0   8.4583        Q  Third   
6         0       1    male  54.0      0      0  51.8625        S  First   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True  
5    man        True  NaN   Queenstown    no   True  
6    man        True    E  Southampton    no   True

### Last Rows Of Data

In [5]:
df.tail(8)

survived  pclass     sex   age  sibsp  parch    fare embarked   class  \
883         0       2    male  28.0      0      0  10.500        S  Second   
884         0       3    male  25.0      0      0   7.050        S   Third   
885         0       3  female  39.0      0      5  29.125        Q   Third   
886         0       2    male  27.0      0      0  13.000        S  Second   
887         1       1  female  19.0      0      0  30.000        S   First   
888         0       3  female   NaN      1      2  23.450        S   Third   
889         1       1    male  26.0      0      0  30.000        C   First   
890         0       3    male  32.0      0      0   7.750        Q   Third   

       who  adult_male deck  embark_town alive  alone  
883    man        True  NaN  Southampton    no   True  
884    man        True  NaN  Southampton    no   True  
885  woman       False  NaN   Queenstown    no  False  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True

### Selecting Sample Rows

In [6]:
df.sample(10)

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
359         1       3  female   NaN      0      0   7.8792        Q   Third   
591         1       1  female  52.0      1      0  78.2667        C   First   
807         0       3  female  18.0      0      0   7.7750        S   Third   
627         1       1  female  21.0      0      0  77.9583        S   First   
185         0       1    male   NaN      0      0  50.0000        S   First   
702         0       3  female  18.0      0      1  14.4542        C   Third   
92          0       1    male  46.0      1      0  61.1750        S   First   
236         0       2    male  44.0      1      0  26.0000        S  Second   
143         0       3    male  19.0      0      0   6.7500        Q   Third   
759         1       1  female  33.0      0      0  86.5000        S   First   

       who  adult_male deck  embark_town alive  alone  
359  woman       False  NaN   Queenstown   yes   True  
591  woman       False    D    Cherbourg   yes  False  
807  woman       False  NaN  Southampton    no   True  
627  woman       False    D  Southampton   yes   True  
185    man        True    A  Southampton    no   True  
702  woman       False  NaN    Cherbourg    no  False  
92     man        True    E  Southampton    no  False  
236    man        True  NaN  Southampton    no  False  
143    man        True  NaN   Queenstown    no   True  
759  woman       False    B  Southampton   yes   True

### Saving DataSet in csv File

In [16]:
df.to_csv("titanic.csv")

TypeError: NDFrame.to_csv() got an unexpected keyword argument 'skipcols'

### Saving DataSet in Excel File

In [13]:
df.to_excel("titanic.xlsx")

### How To Import My Own DataSet

In [25]:
df = pd.read_csv("titanic.csv")
df

Unnamed: 0  survived  pclass     sex   age  sibsp  parch     fare  \
0             0         0       3    male  22.0      1      0   7.2500   
1             1         1       1  female  38.0      1      0  71.2833   
2             2         1       3  female  26.0      0      0   7.9250   
3             3         1       1  female  35.0      1      0  53.1000   
4             4         0       3    male  35.0      0      0   8.0500   
..          ...       ...     ...     ...   ...    ...    ...      ...   
886         886         0       2    male  27.0      0      0  13.0000   
887         887         1       1  female  19.0      0      0  30.0000   
888         888         0       3  female   NaN      1      2  23.4500   
889         889         1       1    male  26.0      0      0  30.0000   
890         890         0       3    male  32.0      0      0   7.7500   

    embarked   class    who  adult_male deck  embark_town alive  alone  
0          S   Third    man        True  NaN  Southampton    no  False  
1          C   First  woman       False    C    Cherbourg   yes  False  
2          S   Third  woman       False  NaN  Southampton   yes   True  
3          S   First  woman       False    C  Southampton   yes  False  
4          S   Third    man        True  NaN  Southampton    no   True  
..       ...     ...    ...         ...  ...          ...   ...    ...  
886        S  Second    man        True  NaN  Southampton    no   True  
887        S   First  woman       False    B  Southampton   yes   True  
888        S   Third  woman       False  NaN  Southampton    no  False  
889        C   First    man        True    C    Cherbourg   yes   True  
890        Q   Third    man        True  NaN   Queenstown    no   True  

[891 rows x 16 columns]

### Dropping *Unnamed* Column

In [27]:
df.columns.str.match("Unnamed")
df = df.loc[:,~df.columns.str.match("Unnamed")]
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

### Checking DataTypes of DataSet

In [28]:
df.dtypes

survived         int64
pclass           int64
sex             object
age            float64
sibsp            int64
parch            int64
fare           float64
embarked        object
class           object
who             object
adult_male        bool
deck            object
embark_town     object
alive           object
alone             bool
dtype: object

### Getting More Information of Data

In [29]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   who          891 non-null    object 
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    object 
 12  embark_town  889 non-null    object 
 13  alive        891 non-null    object 
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), object(7)
memory usage: 92.4+ KB


There null values in *age* and *deck*
### Counting Null Values

In [31]:
df.isnull().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

We can see that there are **177** null values in **age** and **688** in ***deck***

**Population:**
- Population represets every single unit of a subject.
- A descriptive measure for an entire a Population is is its Parameter
- Parameters determine how distribution fits population

**Sample**
- A sepcific group that you will collect from Population
- Sample size is always less than Population Size
-  A descriptive measure for a sample is called Satistics

